xnmt/decoder.py

```diff
@@ -130,6 +136,9 @@ def get_scores(self, mlp_dec_state):
     """
     return self.mlp_layer(dy.concatenate([mlp_dec_state.rnn_state.output(), mlp_dec_state.context]))
 
+  def get_scores_logsoftmax(self, mlp_dec_state):
```

Comment:
What about just `get_scores`? I think it's fine to do `log_softmax` in the parent function if necessary. We might not need to take the log softmax.

Reply:
In the case where we want to modify the probability of the vocabulary directly in the decoder (e.g. linear interpolation), the softmax must be computed in the decoder, not in the parent function.

Reply:
OK, that sounds reasonable.